subprojects/docs/src/docs/userguide/authoring-builds/multi_project_builds.adoc

```diff
@@ -801,6 +905,50 @@ In order to make good use of cross project configuration without running into is
+By default, the configuration of all projects happens before any task is executed.
+This means that when a single task, from a single project is requested, _all_ projects of multi-project build are configured first.
+The reason every project needs to be configured is to support the flexibility of accessing and changing any part of the Gradle project model.
+
+
+[[sec:configuration_on_demand]]
+== Configuration on demand
+
+The _Configuration injection_ feature and access to the complete project model are possible because every project is configured before the execution phase.
+Yet, this approach may not be the most efficient in a very large multi-project build.
+There are Gradle builds with a hierarchy of hundreds of subprojects.
+The configuration time of huge multi-project builds may become noticeable.
+Scalability is an important requirement for Gradle.
+Hence, starting from version 1.4 a new incubating 'configuration on demand' mode is introduced.
+
+Configuration on demand mode attempts to configure only projects that are relevant for requested tasks, i.e. it only executes the `build.gradle[.kts]` file of projects that are participating in the build.
+This way, the configuration time of a large multi-project build can be reduced.
+In the long term, this mode will become the default mode, possibly the only mode for Gradle build execution.
+The configuration on demand feature is incubating so not every build is guaranteed to work correctly.
+The feature should work very well for multi-project builds that have <<#sec:decoupled_projects,decoupled projects>>.
+In “configuration on demand” mode, projects are configured as follows:
+
+* The root project is always configured.
+This way the typical common configuration is supported (allprojects or subprojects script blocks).
+* The project in the directory where the build is executed is also configured, but only when Gradle is executed without any tasks.
+This way the default tasks behave correctly when projects are configured on demand.
+* The standard project dependencies are supported and makes relevant projects configured.
+If project A has a compile dependency on project B then building A causes configuration of both projects.
+* The task dependencies declared via task path are supported and cause relevant projects to be configured.
+Example: `someTask.dependsOn(":some-other-project:someOtherTask")`
+* A task requested via task path from the command line (or Tooling API) causes the relevant project to be configured.
+For example, building 'project-a:project-b:someTask' causes configuration of project-b.
+
+
+Eager to try out this new feature? To configure on demand with every build run see <<build_environment.adoc#sec:gradle_configuration_properties,Gradle properties>>.
+To configure on demand just for a given build, see <<command_line_interface.adoc#sec:command_line_performance,command-line performance-oriented options>>.
```

Comment:
Those two sections I just moved down (and added line breaks to not have to scroll sideways when working on it) as they felt out-of-place in middle of cross-configuration section.
I'm not sure how relevant or accurate this information is today, but that's another story.